src/System.IO.FileSystem/tests/File/ReadWriteAllBytes.cs

```diff
@@ -55,6 +55,30 @@ public void ValidWrite(int size)
         }
 
         [Fact]
+        [OuterLoop]
+        public void ReadFileOver2GB()
+        {
+            string path = GetTestFilePath();
+            using (var writer = new StreamWriter(File.Create(path)))
+            {
+                int split = 10;
+                for (int i = 0; i < (split + 1); i++)
+                {
+                    try
+                    {
+                        writer.Write(new string('a', Int32.MaxValue / split));
+                        writer.Flush();
+                    }
+                    catch (OutOfMemoryException)
+                    {
+                        split = split / 2;
+                    }
+                }
+            }
+            Assert.Throws<IOException>(() => File.ReadAllBytes(path));
```

Comment:
Might be nice to verify that we can subsequenty open the file and read it in smaller chunks with a file stream, demonstrating that we can read large files but we fail in this case because it's too many elements for a single array.